src/amo/pages/Guides/index.js

```diff
@@ -119,14 +120,16 @@ export class GuidesBase extends React.Component<InternalProps> {
   constructor(props: InternalProps) {
     super(props);
 
-    const { errorHandler, guids } = this.props;
+    const { addons, errorHandler, guids, loading } = this.props;
 
-    this.props.dispatch(
-      fetchGuidesAddons({
-        guids,
-        errorHandlerId: errorHandler.id,
-      }),
-    );
+    if (!loading && Object.keys(addons).length === 0) {
```

Comment:
We should think about the case where no addons are returned by the API. This code will try to always re-fetch the add-ons.

I believe you could make `addons` nullable in the Flow type, and change a bit the logic in `mapStateToProps()` so that either we have a map of `addons` or `null`. Or maybe, you could always initialize the map with the result of `getAddonByGUID()`.